bin/oref0-pump-loop.sh

```diff
@@ -262,15 +254,15 @@ function smb_verify_reservoir {
     echo -n "Checking reservoir: " \
     && ( check_reservoir || check_reservoir ) 2>&3 >&4 \
     && echo -n "reservoir level before: " \
-    && cat monitor/lastreservoir.json | tr -d '\n' \
+    && cat monitor/lastreservoir.json | nonl \
     && echo -n ", suggested: " \
-    && jq -r -C -c .reservoir enact/smb-suggested.json | tr -d '\n' \
+    && jq -r -C -c .reservoir enact/smb-suggested.json | nonl \
     && echo -n " and after: " \
     && cat monitor/reservoir.json \
     && (( $(bc <<< "$(< monitor/lastreservoir.json) - $(< monitor/reservoir.json) <= 0.1") )) \
     && (( $(bc <<< "$(< monitor/lastreservoir.json) - $(< monitor/reservoir.json) >= 0") )) \
-    && (( $(bc <<< "$(jq -r .reservoir enact/smb-suggested.json | tr -d '\n') - $(< monitor/reservoir.json) <= 0.1") )) \
-    && (( $(bc <<< "$(jq -r .reservoir enact/smb-suggested.json | tr -d '\n') - $(< monitor/reservoir.json) >= 0") ))
+    && (( $(bc <<< "$(jq -r .reservoir enact/smb-suggested.json | nonl) - $(< monitor/reservoir.json) <= 0.1") )) \
```

Comment:
I might be wrong, but i think we need `bc -l` for decimal calculations?

Reply:
`bc` uses whatever precision/number of decimal places are in its input. This is actually pretty sketchy; for example `bc <<< "0.5 * 0.5` outputs `0.2`. But in this case it seems to work out okay.

The `-l` option is defined in bc's manpage as "Define the standard math library." So I think we only need this for functions like cosine.

I'm open to suggestions for things to replace `bc` with for simple floating point arithmetic that are usable from shell. Unfortunately `node` won't work, because its interpreter startup time is too slow (about 800ms on my Edison, vs 10ms for bc).

Reply:
If the version of jq you have includes slurpfile or argfile support, you could theoretically cut the overhead of two or three of the three or four subshells being invoked per line here by replacing the bc calls with jq calls similar to:
```
jq -r "if (.reservoir - \$monitor[] <= 0.1) then 1 else 0 end" --slurpfile monitor monitor/reservoir.json enact/smb-suggested.json
```